src/fiatExchanges/SelectProvider.tsx

```diff
@@ -156,7 +173,11 @@ export default function SelectProviderScreen({ route, navigation }: Props) {
         cryptoAmount={route.params.amount.crypto}
         coinbaseProvider={coinbaseProvider}
       />
-      <ExchangesSection selectedCurrency={route.params.selectedCrypto} flow={flow} />
+      <ExchangesSection
+        numExchanges={exchanges?.length}
```

Comment:
Since we're actually fetching all the exchanges here, we should probably avoid fetching them again elsewhere. With this implementation, we fetch exchanges once here, and then, if a user actually browses to the Exchanges screen by clicking the card (assuming some exchanges are available), the list of exchanges will be fetched again. I would consider passing the actual exchange information (rather than just the number of exchanges) down into the `ExchangesSection` component, and further into the `ExternalExchanges` screen itself, and remove the `fetchExchanges` call from within `ExternalExchanges`.